### DISCUSSION TITLE
Redis auth and tls

### DISCUSSION THREAD

**Discussion by idenkov** [vectordotdev/vector#16479] [Q&A] [answered]
Hello,

I am having trouble to understand how to configure connection to redis cluster with tls and auth enabled.
I looked into the documentation, but did not found anything about specifying auth and tls certs.

Thanks!

**Comment by neuronull** [comment, marked as answer]
Hello,

I believe the way to authenticate is through the `url` option, and using `rediss` for TLS.

https://vector.dev/docs/reference/configuration/sources/redis/#url

For example:

```
rediss://some_user:some_password@localhost:1234/some_db
```

**Reply by idenkov** [reply]
Hi, 

Thank you, I forgot you can specify user:pass in the url, however I still need to send at least CA cert to connect to TLS enabled clusters.

**Reply by neuronull** [reply]
Hmmm it looks like we use the crate`redis-rs` to interface with Redis.
And I'm seeing an open issue tracking the support for allowing more flexible TLS settings: https://github.com/redis-rs/redis-rs/issues/463

AFAICT that feature has not been added to that crate, and since vector is depending on that for the Redis connection, I don't think we'd be able to work around that, aside from stepping away from `redis-rs` (whether to another crate or to something home rolled).

I have not seen a GH issue in Vector asking for this yet, so in the meantime, I would recommend opening a GH issue in Vector for this feature request, and we can mention their this limitation of `redis-rs`.

**Reply by idenkov** [reply]
Thank you, I also just noticed that it might not support redis cluster as well, when Vector is trying to write its key to the instance it gets `MOVED` which is standard response from a redis cluster that means it is on different node on the cluster, and the client should get it from the returned nodes. Also I can't find anything related to redis cluster in the documentation. So I guess redis cluster is not supported? Should I open two GH issues one for cluster supports and one for TLS?

```Feb 16 21:54:25 rediscluster001 vector[32084]: 2023-02-16T21:54:25.501292Z ERROR source{component_kind="source" component_id=redis_metrics component_type=redis component_name=redis_metrics}: vector::internal_events::redis: Failed to read message. error=An error was signalled by the server: 12785 [10.177.14.192:6379](http://10.177.14.192:6379/) error_code=MOVED error_type="reader_failed" stage="sending" internal_log_rate_limit=true```

**Reply by neuronull** [reply]
You are welcome.

Oh interesting. Yeah I'm not finding other record of the redis cluster being brought up in Vector channels either.
Yes, two separate issues would be best. Thank you.
